### PR TITLE
cache: simplify indexing funcs

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

### CODE REVIEW

File: pkg/cache/controller.go [hunk 1]
```diff
@@ -37,8 +36,12 @@ func ClusterIndexFunc(obj interface{}) ([]string, error) {
 	if err != nil {
 		return []string{}, fmt.Errorf("object has no meta: %v", err)
 	}
-	clusterName := logicalcluster.From(meta).String()
-	return []string{ToClusterAwareKey(clusterName, "", "")}, nil
+	return []string{ClusterIndexKey(logicalcluster.From(meta))}, nil
+}
+
+// ClusterIndexKey formats the index key for a cluster name
+func ClusterIndexKey(clusterName logicalcluster.Name) string {
```
Comment (stevekuznetsov) [inline pkg/cache/controller.go:43]: Exposing a clear index-formatter func allows the user to know exactly what types are required to use the index.

File: pkg/cache/controller_test.go [hunk 2]
```diff
@@ -21,29 +21,38 @@ import (
...
-
 func TestClusterIndexFunc(t *testing.T) {
 	tests := map[string]struct {
-		obj     *unstructured.Unstructured
+		obj     *metav1.ObjectMeta
 		desired string
 	}{
-		"bare cluster":             {obj: newUnstructured("test", "", "name", nil), desired: "test//"},
-		"cluster and namespace":    {obj: newUnstructured("test", "namespace", "name", nil), desired: "test//"},
-		"bare cluster with dashes": {obj: newUnstructured("test-with-dashes", "", "name", nil), desired: "test-with-dashes//"},
```
Comment (stevekuznetsov) [inline pkg/cache/controller_test.go:46]: This is a personal nit but having to read helpers to figure out what the order of these arguments means, along with having to figure out that unstructured & `nil` labels were red herrings, is not ideal for comprehension of a test IMO

File: pkg/cache/keyfunc.go [hunk 3]
```diff
@@ -48,15 +48,20 @@ func MetaClusterNamespaceKeyFunc(obj interface{}) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("object has no meta: %v", err)
 	}
+	return ToClusterAwareKey(logicalcluster.From(meta).String(), meta.GetNamespace(), meta.GetName()), nil
+}
+
+// ToClusterAwareKey is a helper function that formats cluster, namespace, and name for key and index functions
```
Comment (ncdc) [inline pkg/cache/keyfunc.go:54]: this is just for key functions, right?

Comment (stevekuznetsov) [reply pkg/cache/keyfunc.go]: Yep, sorry - updated

**Review by ncdc — APPROVED** [review]

**Pull request merged**